### PR TITLE
Deprecate Python 2

### LINKED ISSUE #706: Remove support for Python 2
Add Python 3.7 to the Travis and AppVeyor test matrix. Deprecate Python 2.7 support in v1.0.

And while we're at it - let's add CI posts to Slack.

### PR DESCRIPTION
**Pull request by bbengfort**
This PR deprecates YB support for Python 2 by removing py2 tests from our CI and attempting to remove any py2 specific tests.

Fixes #706

Still todo:

- [ ] Travis-CI not sending notifications to Slack?
- [x] Confirm AppVeyor notifications to Slack 
- [x] Find any py2-specific tests and remove them 
- [ ] Should we add Python 3.5 test support?

### CODE REVIEW

**Comment by bbengfort** [comment]
I think this is ready for review:

1. Updated Travis to test Python 3.6 and 3.7 using the new xenial image
2. Updated AppVeyor to test Python 3.6, 3.7, and Miniconda 3.6 
3. Removed all references to `six` for py2to3 compatibility
4. Removed import trap for `unittest.mock` 
5. Removed references to `basestring` and `xrange` 
6. Removed all `__future__` imports

For the Python 3.5 question, we will have to change all `assert_called_once` statements to `assert_called_once_with` statements ... my $0.02 is that testing 3.6 and 3.7 should be sufficient and Python 3.8 will probably come out soon anyway. We don't want CI to take forever. 

As for Travis - I've tried a number of things; but I'm wondering if the issue is because the Travis configuration is not in the `develop` branch - I may have to work directly on the `develop` branch to get this taken care of, either way a merge for this PR is needed to sort that out.

File: yellowbrick/style/palettes.py [hunk 1]
```diff
@@ -20,16 +20,12 @@
...
-from __future__ import division
-
 import numpy as np
 import matplotlib as mpl
 import matplotlib.pyplot as plt
 import matplotlib.colors as mplcol
 
 from itertools import cycle
-from six import string_types
-from six.moves import range
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply yellowbrick/style/palettes.py]: It was - in Python 2 `range` is not a generator but returns an in-memory list whereas `xrange` returns a memory safe generator. In Python 3, `range` is Python2 `xrange` - so this import from `six` ensure the same behavior when using `range`.

**Comment by bbengfort** [comment]
@lwgray thank you for your review! Since this is such a big change, I hope you won't mind if we also include additional reviews? I want to ensure we're all on the same page moving forward with Python 3 only -- @DistrictDataLabs/team-oz-maintainers any other comments/concerns?

File: yellowbrick/style/palettes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lwgray) [reply yellowbrick/style/palettes.py]: thanks...I really appreciate it..

**Comment by lwgray** [comment]
I completely and utterly understand, Please have others review this.  I had no expectation that mine would be the only review.

File: yellowbrick/datasets/download.py [hunk 2]
```diff
@@ -17,13 +17,8 @@
...
 
 import os
-import six
 import zipfile
 
-if six.PY2:
-    # backport for encoding in open for python2
-    from io import open
-
 try:
```
Comment (rebeccabilbro) [inline yellowbrick/datasets/download.py]: Great catch on the six stuff! In addition to dropping that, I think we can also safely change the try-except `urllib` import in this same file to just `from urllib.request import urlopen`

File: yellowbrick/utils/timer.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ class Timer:
     A context object timer. Usage:
         >>> with Timer() as timer:
         ...     do_something()
-        >>> print timer.interval
```
Comment (rebeccabilbro) [inline yellowbrick/utils/timer.py]: Thanks for updating this example!

**Comment by bbengfort** [comment]
@rebeccabilbro yes, that would be great if you could update the urllib import before merging in, nice catch and thank you!

**Pull request merged**